frontend/src/v5/validation/containerAndFederationSchemes/validators.ts

```diff
@@ -18,45 +18,55 @@
 import * as Yup from 'yup';
 import { formatMessage } from '@/v5/services/intl';
 
-export const numberField = Yup.number().typeError(formatMessage({
-	id: 'settings.surveyPoint.error.number',
-	defaultMessage: 'Must be a decimal number or integer',
-}));
+export const numberField = Yup.number()
+	.default(0)
+	.typeError(formatMessage({
+		id: 'settings.surveyPoint.error.number',
+		defaultMessage: 'Must be a decimal number or integer',
+	}));
 
-export const containerName = Yup.string()
+export const name = Yup.string()
 	.max(120,
 		formatMessage({
-			id: 'validation.containers.name.error.max',
-			defaultMessage: 'Container Name is limited to 120 characters',
+			id: 'validation.model.name.error.max',
+			defaultMessage: 'Name is limited to 120 characters',
 		}))
 	.required(
 		formatMessage({
-			id: 'validation.containers.name.error.required',
-			defaultMessage: 'Container Name is a required field',
+			id: 'validation.model.name.error.required',
+			defaultMessage: 'Name is a required field',
 		}),
 	);
 
-export const containerUnit = Yup.string().required().oneOf(['mm', 'cm', 'dm', 'm', 'ft']).default('mm');
-export const containerType = Yup.string().required().default('Uncategorised');
+export const unit = Yup.string().required().oneOf(['mm', 'cm', 'dm', 'm', 'ft']).default('mm');
+export const type = Yup.string().required().default('Uncategorised');
 
-export const containerCode = Yup.string()
-	.max(50,
-		formatMessage({
-			id: 'validation.containers.code.error.max',
-			defaultMessage: 'Code is limited to 50 characters',
-		}))
-	.matches(/^[\w|_|-]*$/,
-		formatMessage({
-			id: 'validation.containers.code.error.characters',
-			defaultMessage: 'Code can only consist of letters, numbers, hyphens or underscores',
-		}));
+export const code = Yup.lazy((value) => (
+	value === ''
+		? Yup.string().strip()
+		: Yup.string()
```

Comment:
When the value is empty, we should filter it out so that we don't pass it to the backend.
What happens here is the value is stripped if it is an empty string https://github.com/jquense/yup/issues/262#issuecomment-408453019

Reply:
ahh,  then  I think we could create a function to reuse it.  Somthing like 

```suggestion

const stripIfBlank = (value) => (
	value === ''
		? Yup.string().strip()
		: Yup.string());

export const code = Yup.lazy((value) => (
	stripIfBlank(value)
	.max(50,
```